department-of-veterans-affairs/vets-api
# 64194 betamocks map sus

## Summary

- This PR adds the MAP STS endpoint details to the betamocks configuration.
- This facilitates for rapid development for ToU and other features interacting with MAP STS API; otherwise, development requires use of SOCKS proxy.
- This work is being completed by @vsp-identity

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#64191

## Testing done

Prior to this change, any call to the STS server would fail. This PR enables betamocks for these calls, so they should succeed and return a mocked response.

### Test Steps

- Pull the latest changes to the `vets-api-mockdata` repo.
  - Changes should include additions from PR https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/420
  - If this PR isn't merged yet, check out the branch locally.
- Open a rails console.
- Require the SUS service.
```ruby
require 'mobile_application_platform/sign_up/service'
```
- Call the service different methods with the required parameters:
```ruby
MobileApplicationPlatform::SignUp::Service.new.status(icn:)
MobileApplicationPlatform::SignUp::Service.new.agreements_accept(icn:, signature_name:, version:)
MobileApplicationPlatform::SignUp::Service.new.agreements_decline(icn:)
MobileApplicationPlatform::SignUp::Service.new.update_provisioning(icn:, first_name:, last_name:, mpi_gcids:)
```
- Verify the response is mocked.

## Screenshots

Not relevant.

## What areas of the site does it impact?

This does not have an impact to production facilities of the site; it only affects local development.

## Acceptance criteria


## Requested Feedback

Please advise on best practices using the betamocks library and whether additional configuration is necessary.
